Diakonie 554 morning email reminder failling

Leandro13Silva13 opened the pull request.

Leandro13Silva13 committed:
feat: DIAKONIE-554
Adapt the sendEmailNotificationsForOpenEnquiries cron to also send the tenant name and claim

Leandro13Silva13 committed:
fix: DIAKONIE-554
run mvn spotless:apply

Leandro13Silva13 committed:
fix: DIAKONIE-554
Update needed tests

Leandro13Silva13 committed:
fix: DIAKONIE-554
Remove incorrect code

Leandro13Silva13 committed:
fix: DIAKONIE-554
Fix test because of the wrong code removal

Leandro13Silva13 committed:
fix: DIAKONIE-554
Update the code to follow the correct logic and update needed tests

vi-sudo commented:
Task linked: [OB-21465 [Diakonie] Bug: Morning Email Reminder Failling on PROD](https://app.clickup.com/t/6656196/OB-21465)

roman-vi commented on src/main/java/de/caritas/cob/userservice/api/workflow/enquirynotification/scheduler/EnquiryNotificationScheduler.java:
```diff
@@ -19,10 +20,13 @@ public class EnquiryNotificationScheduler {
   @Value("${enquiry.open.notification.enabled}")
   private Boolean enquiryNotificationsEnabled;
 
+  private final @NonNull TenantContextProvider tenantContextProvider;
```
I would rather refactor a little in TenantContextProvider to use TECHNICAL_TENANT_ID from TenantContext and not from TenantResolverService

Leandro13Silva13 replied:
I am not understanding the comment sorry..

roman-vi commented on src/main/java/de/caritas/cob/userservice/api/workflow/enquirynotification/service/EnquiryNotificationService.java:
```diff
@@ -49,6 +50,10 @@ public class EnquiryNotificationService {
   private final @NonNull ReleaseToggleService releaseToggleService;
+  private final TenantTemplateSupplier tenantTemplateSupplier;
+
+  @Value("${multitenancy.enabled}")
```
Why having it 16 times in the code and not wrapping in single configuration bean which provides the flag? We can combine multitenancy related options there e.g., multitenancyWithSingleDomain

Leandro13Silva13 replied:
I agree, but I am already overtime on this task, and this would imply more effort on development and tests. We can discuss this on the daily tomorrow :)

roman-vi replied:
agreed to discuss in daily first

roman-vi commented on src/test/java/de/caritas/cob/userservice/api/workflow/enquirynotification/scheduler/EnquiryNotificationSchedulerTest.java:
```diff
@@ -21,6 +22,8 @@ class EnquiryNotificationSchedulerTest {
   @Mock private EnquiryNotificationService enquiryNotificationService;
 
+  @Mock private TenantContextProvider tenantContextProvider;
+
```
please add into `sendEmailNotificationsForOpenEnquiries_Should_callEnquiryNotificationService_When_featureIsEnabled` verify `tenantContextProvider` is called

Leandro13Silva13 committed:
fix: DIAKONIE-554
fix sendEmailNotificationsForOpenEnquiries_Should_callEnquiryNotificationService_When_featureIsEnabled test, after PR comments

Leandro13Silva13 committed:
fix: DIAKONIE-554
remove TECHNICAL_TENANT_ID from TenantResolverService, so we only have 1 on TenantContextProvider
